src/python/zquantum/core/circuit/_circuit_test.py

```diff
@@ -96,13 +123,124 @@ def test_circuit_init(self):
 
         self.assertEqual(circuit_from_cirq, circuit_from_pyquil)
 
+    def test_circuit_symbolic_params(self):
+        # Given
+        qubits = [Qubit(i) for i in range(0, 3)]
+        theta_1 = sympy.Symbol("theta_1")
+        theta_2 = sympy.Symbol("theta_2")
+
+        gate_1 = Gate("Rx", params=[theta_1], qubits=[qubits[0]])
+        gate_2 = Gate("Ry", params=[theta_2], qubits=[qubits[0]])
+        gate_3 = Gate("Rz", params=[0.5 * theta_1 - 2 * theta_2], qubits=[qubits[0]])
+        gate_4 = Gate("Rx", params=[0.3], qubits=[qubits[0]])
+
+        target_symbolic_params_1 = [theta_1, theta_2]
+        target_symbolic_params_2 = []
+
+        circ_1 = Circuit("circ1")
+        circ_1.qubits = qubits
+        circ_1.gates = [gate_1, gate_2, gate_3, gate_4]
+
+        circ_2 = Circuit("circ2")
+        circ_2.qubits = qubits
+        circ_2.gates = [gate_4]
+
+        # When
+        symbolic_params_1 = circ_1.symbolic_params
+        symbolic_params_2 = circ_2.symbolic_params
+
+        # Then
+        self.assertEqual(set(symbolic_params_1), set(target_symbolic_params_1))
+        self.assertEqual(symbolic_params_2, target_symbolic_params_2)
```

Comment:
This seems more like two separate unittests -> circuit with no params, circuit with some params. Although I don't think it's necessary to change

Reply:
Naming is confusing – method is called `symbolic_params` and it's method of `circuit` and it's `test`, so altogether it's `test_circuit_symbolic_params`, but the test actually tests it for circuit with and without symbolic params, to make sure the behaviour is correct if circuit is "regular".

Reply:
Good point - maybe these should actually be assertions in `test_circuit_init` and `test_circuit_init_with_symbolic_params`?

Reply:
I've updated the tests and added a warning, see if it makes more sense now.